### ISSUE TITLE
[Feat]: cull_idle_timeout mechanism requirement

### ISSUE THREAD

**Issue by ycsong1212** [coder/code-server#5008]
## What is your suggestion?
Maybe cull_idle_timeout mechanism should be provided. If there hasn't been an active connection to code-server in X minutes, we should exit with status 0. It has been mentioned in #1636 @nhooyr 

## Why do you want this feature?
I am working on develop online VSCode support by code-server. To enhance the resource utilization rate, I require cull-idle-timeout mechanism. If users do not use for a period of time, I hope the container killed. According to the doc, code-server provides the heartbeat mechanism. If I keep the WebPage on but do nothing, the time of heartbeat will be updated every minute. This is not what I want.

## Others
I have seen that such feature is referred like issue #1636. However, I failed to launch with --idle-timeout args. I want to confirm if  such function is supported right now? If not, is there any methods to support my requirement? I'll appreciate it if someone could help.

**Comment by jsjoeio** [comment]
Please don't tag [nhooyr](https://github.com/nhooyr) - they are no longer working on the project. 

Thank you for the feature request! We're not adding new features at this time but we'll track it in the Backlog regardless.

**Comment by ycsong1212** [comment]
> Please don't tag [nhooyr](https://github.com/nhooyr) - they are no longer working on the project.
> 
> Thank you for the feature request! We're not adding new features at this time but we'll track it in the Backlog regardless.

Thanks for reply, I am looking forward to this new feature. Could you please change the label to feature instead of enhancement? I think this is wrongly classified and it seems that I cannot change it myself.

**Comment by code-asher** [comment]
Whoever works on this make sure it works with `--enable-remote-auto-shutdown` and `--remote-auto-shutdown-without-delay` from VS Code.

**Comment by ycsong1212** [comment]
> enable-remote-auto-shutdown

I have searched these two params but I failed to find how to use them. Are they supported in the newest version 4.12.0? If so, could u plz provide an example to use these two params? Thx~

**Comment by code-asher** [comment]
No, these are not supported in any version code-server currently.
